cli/cliui/cliui.go

```diff
@@ -117,48 +101,150 @@ func ifTerm(fmt pretty.Formatter) pretty.Formatter {
 }
 
 func init() {
+	color = termenv.NewOutput(os.Stdout).ColorProfile()
+	if flag.Lookup("test.v") != nil {
+		// Use a consistent colorless profile in tests so that results
+		// are deterministic.
+		color = termenv.Ascii
+	}
+
 	// We do not adapt the color based on whether the terminal is light or dark.
 	// Doing so would require a round-trip between the program and the terminal
 	// due to the OSC query and response.
-	DefaultStyles = Styles{
-		Code: pretty.Style{
-			ifTerm(pretty.XPad(1, 1)),
-			pretty.FgColor(Red),
-			pretty.BgColor(color.Color("#2c2c2c")),
-		},
-		DateTimeStamp: pretty.Style{
-			pretty.FgColor(color.Color("#7571F9")),
-		},
-		Error: pretty.Style{
-			pretty.FgColor(Red),
-		},
-		Field: pretty.Style{
-			pretty.XPad(1, 1),
-			pretty.FgColor(color.Color("#FFFFFF")),
-			pretty.BgColor(color.Color("#2b2a2a")),
-		},
-		Keyword: pretty.Style{
-			pretty.FgColor(Green),
-		},
-		Placeholder: pretty.Style{
-			pretty.FgColor(color.Color("#4d46b3")),
-		},
-		Prompt: pretty.Style{
-			pretty.FgColor(color.Color("#5C5C5C")),
-			pretty.Wrap("> ", ""),
-		},
-		Warn: pretty.Style{
-			pretty.FgColor(Yellow),
+	DefaultStyles = Styles{Wrap: pretty.Style{pretty.LineWrap(80)}}
+
+	defaultTheme := userThemeConfig{
+		Colors: map[string]string{
+			"green":  "#04B575",
+			"red":    "#ED567A",
+			"yellow": "#ECFD65",
+			"blue":   "#5000ff",
 		},
-		Wrap: pretty.Style{
-			pretty.LineWrap(80),
+		Styles: &userThemeStyles{
+			Code: &userThemeStyle{
+				FgColor: "$red",
+				BgColor: "#2C2C2C",
+			},
+			DateTimeStamp: &userThemeStyle{
+				FgColor: "#7571F9",
+			},
+			Error: &userThemeStyle{
+				FgColor: "$red",
+			},
+			Field: &userThemeStyle{
+				FgColor: "#FFFFFF",
+				BgColor: "#2B2A2A",
+			},
+			FocusedPrompt: &userThemeStyle{
+				FgColor: "$blue",
+			},
+			Keyword: &userThemeStyle{
+				FgColor: "$green",
+			},
+			Placeholder: &userThemeStyle{
+				FgColor: "#4d46b3",
+			},
+			Prompt: &userThemeStyle{
+				FgColor: "#5C5C5C",
+			},
+			Warn: &userThemeStyle{
+				FgColor: "$yellow",
+			},
 		},
 	}
 
-	DefaultStyles.FocusedPrompt = append(
-		DefaultStyles.Prompt,
-		pretty.FgColor(Blue),
-	)
+	configDir := config.DefaultDir()
+	if dir := os.Getenv("CODER_CONFIG_DIR"); dir != "" {
+		configDir = dir
+	}
+
+	if theme, err := os.ReadFile(path.Join(configDir, "theme.toml")); err == nil {
+		_ = toml.Unmarshal(theme, &defaultTheme)
+	}
+
+	applyUserTheme(defaultTheme)
```

Comment:
Would it make sense to write the merged theme back to disk?
This would mean that running the binary for the first time will generate a sample configuration that users can then edit as required.

Reply:
I had originally considered this. I realised that a potential issue is that it wouldn't preserve any comments in the file. We may consider that behaviour to be fine but it is something to consider.

Reply:
That's a good point. I could imagine that being frustrating for users.